From 28d375897fe1cec91e74f52ee993890d4a3b4310 Mon Sep 17 00:00:00 2001
From: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Date: Wed, 20 Dec 2017 14:53:51 +0900
Subject: [PATCH 569/909] arm64: dts: r8a77965: Add DEVFREQ support for GSX

commit b2b27be32244216260cf56d3a84807259725f5b9 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

This patch supports GSX frequency scaling using devfreq.

Based on a similar patch of the R8A7796 device tree
by Gaku Inami <gaku.inami.xw@bp.renesas.com>.

Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/renesas/r8a77965.dtsi | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a77965.dtsi b/arch/arm64/boot/dts/renesas/r8a77965.dtsi
index a5186e5cac02..c1ed332116ec 100644
--- a/arch/arm64/boot/dts/renesas/r8a77965.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a77965.dtsi
@@ -168,6 +168,27 @@
 		clock-frequency = <0>;
 	};
 
+	gsx_opp_table: gsx_opp_table {
+		compatible = "operating-points-v2";
+
+		opp-200000000 {
+			opp-hz = /bits/ 64 <200000000>;
+			opp-microvolt = <830000>;
+		};
+		opp-300000000 {
+			opp-hz = /bits/ 64 <300000000>;
+			opp-microvolt = <830000>;
+		};
+		opp-400000000 {
+			opp-hz = /bits/ 64 <400000000>;
+			opp-microvolt = <830000>;
+		};
+		opp-600000000 {
+			opp-hz = /bits/ 64 <600000000>;
+			opp-microvolt = <830000>;
+		};
+	};
+
 	/* MSIOF reference clock - to be overridden by boards that provide it */
 	msiof_ref_clk: msiof-ref-clock {
 		compatible = "fixed-clock";
@@ -1909,6 +1930,7 @@
 			reg = <0 0xfd000000 0 0x40000>;
 			interrupts = <GIC_SPI 119 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&cpg CPG_MOD 112>;
+			operating-points-v2 = <&gsx_opp_table>;
 			power-domains = <&sysc R8A77965_PD_3DG_B>;
 			resets = <&cpg 112>;
 		};
-- 
2.17.1

